solr/core/src/java/org/apache/solr/api/ContainerPluginsRegistry.java

```diff
@@ -344,7 +344,7 @@ public ApiInfo(PluginMetaHolder infoHolder, List<String> errs) {
         }
       } else {
         try {
-          klas = Class.forName(klassInfo.className);
+          klas = Class.forName(klassInfo.className, true, coreContainer.getResourceLoader().getClassLoader());
```

Comment:
the ClassLoader inside SolrResourceLoader is an implementation detail; SRL can find the class via `findClass`.
`coreContainer.getResourceLoader().findClass(.klassInfo.className,Object.class)`